libparsec/crates/types/src/local_manifest.rs

```diff
@@ -336,33 +370,41 @@ impl LocalFileManifest {
             blocks,
             base: remote,
         };
-        // TODO: use proper error handling
-        manifest.assert_integrity();
+
+        // Remote manifests comes from the certificate ops, so they are expected
+        // to be validated using `CertifOps::validate_child_manifest`.
+        // However, we still check the content integrity of the local manifest just in case.
+        manifest
+            .check_content_integrity()
+            .expect("Manifest integrity");
         manifest
     }
 
-    pub fn to_remote(
-        &self,
-        author: DeviceID,
-        timestamp: DateTime,
-    ) -> Result<FileManifest, &'static str> {
-        self.assert_integrity();
+    pub fn to_remote(&self, author: DeviceID, timestamp: DateTime) -> DataResult<FileManifest> {
```

Comment:
I think we should return another error type than `DataError` here:
- the local manifest we work on is expected to be valid (hence if it integrity is wrong we should just panic)
- same thing for the remote manifest we have just generated
- in the end, the only error possible is if the local manifest is not reshaped

On top of that it would be good to avoid adding to `DataError` such a specific error (given the errors in `DataError` are themselves rather broad and related check done during deserialization, while here we are not doing deserialization...)

```rust
#[derive(Debug, thiserror::Error)]
pub enum LocalFileManifestToRemoteError {
  #[error("Local file manifest needs reshape")]
  NeedReshape,
}
```